Sources/FoundationInternationalization/Formatting/Number/ICULegacyNumberFormatter.swift

```diff
@@ -123,7 +123,7 @@ internal final class ICULegacyNumberFormatter : @unchecked Sendable {
     enum NumberFormatType : Hashable, Codable {
         case number(NumberFormatStyleConfiguration.Collection)
         case percent(NumberFormatStyleConfiguration.Collection)
-        case currency(CurrencyFormatStyleConfiguration.Collection)
+        case currency(CurrencyFormatStyleConfiguration.Collection, /*currency code*/ String)
```

Comment:
This type is part of the `Codable` format of a number of public types conforming to `ParseStrategy`, e.g. `IntegerParseStrategy`. So while this solution means the Codable conformance for `Decimal.FormatStyle.Currency` is backwards compatible, your decoding test would still fail for `IntegerParseStrategy`.

Reply:
> This type is part of the Codable format of a number of public types conforming to ParseStrategy, e.g. IntegerParseStrategy

I'm not sure what you meant by this. This is part of `ICULegacyNumberFormatter`, and I don't think we're encoding this type anywhere, unless I missed something?

What's a test that would fail because of this change?

Reply:
Ah, you meant this is a var in `FloatingPointParseStrategy`... OK we need test for that too.

Reply:
- [ ] Add codable test for `ParseStrategy`

Reply:
Another attempt: In [9a1e617](https://github.com/swiftlang/swift-foundation/pull/1074/commits/9a1e617c6affea242c994cdbf45fddb686f6115b) I removed the stored property `numberFormatType` from `IntegerParseStrategy` and `FloatingPointParseStrategy`. That means the we no longer preserve `numberFormatType` in serialization, and `numberFormatType` in previous `Codable` representation will be ignored from now on. 

This is ok because the information is already provided by the other variable, `formatStyle`. In other words, `numberFormatType` has always been redundant. I also liked this because the previous serialization output feels like we're leaking internal information. 

There is indeed a slight chance of regression -- if you manually craft an archive where `numberFormatType` represents different configurations from that of `formatStyle`, previously we'd favor what's represented by `numberFormatType`, but now we'd favor `formatStyle`. I think this risk is low enough though. Besides, I don't think we ever guarantee compatibility for manually crafted archives.